mozdownload/scraper.py

```diff
@@ -242,6 +243,51 @@ def detect_platform(self):
         else:
             return "%s%d" % (mozinfo.os, mozinfo.bits)
 
+    def get_checksum_url(self):
+        """Get the checksum url for current version."""
+        filename = 'SHA512SUMS'
+        url = os.path.split(os.path.split(os.path.split(self.url)[0])[0])[0] + '/' + filename
+        return url
+
+    def has_checksum(self, url):
+        """Check if the specified checksum file exists."""
+        self.logger.info('Checking checksum at url: %s' % url)
+        r = self.session.get(url, stream=True)
+        code = r.status_code
+        return code == 200
+
+    def download_checksum(self):
+        """Download the checksum."""
+        r = self.session.get(self.curl, stream=True)
+        r.raise_for_status()
+
+        with open(self.cfilename, 'wb') as f:
+                for chunk in r.iter_content(CHUNK_SIZE):
+                    f.write(chunk)
```

Comment:
I don't see a reason why we have to write the content of the checksum file to a local file. Just keep it in memory which is also way faster.